### DISCUSSION TITLE
Polar Stereographic Projection

### DISCUSSION THREAD

**Discussion by gtatarsky-IB** [Unidata/netcdf-java#1339] [Questions and Answers]
Hello! I am an engineer working on a NWS project and using this library. I am currently migrating some old python code into java persistence code, and using the 5.x branch of this library. I am wondering why in Grib2Gds, the Polar Stereographic - as opposed to the Lambert, Mercator, and Lat/Lon classes—does not have public fields? For example, LA/LO 1.

In the others:
```
public final float la1;
public final float lo1;
```
In Polar:
```
final float la1;
final float lo1;
```
I am new to this weather code world. I am wondering why this was done, or if it is oversight. I am persisting a lot of different GDS based coverages to my db, and need to be able to access these fields. An easy change—but I want to see why this is the way it is before any attempt at change.

**Comment by gtatarsky-IB** [comment]
I guess several of the classes have non-public vars. Given a Grib2Gds of the stereo and lambert types, and a need to persist their specific fields based on their tables (referenced in the Grib2Gds.java file above each class), what is the recommended way of accessing that data? 
I am able to access everything I need for Lat/Lon and Mercator, but am wondering how I should access the following with minimal overhead:
Polar Stereo and Lambert:
```
final float la1;
final float lo1;
final float lov;
final float lad;
final float dX;
final float dY;
final int flags;
final int projCenterFlag;
```
And for Lambert only:
```
final float latin1;
final float latin2;
final float latSouthPole;
final float lonSouthPole;
```

**Comment by tdrwenski** [comment]
I am not very familiar with the code or why some members were made public. In general, we prefer to have private members to limit unneeded access. If you just need read access to these members, I would suggest adding getter methods and keeping the members themselves package private as they are now.

**Comment by gtatarsky-IB** [comment]
Thanks @tdrwenski — I went with a Horizontal System created from the projections which provided everything I needed (right now) to create a coverage. If I need access to the other values, I agree that a getter would be great! Thanks again—I'll close this.
